startup/27-vstream-cam.py

```diff
@@ -69,15 +71,30 @@ def trigger(self, *args, **kwargs):
             resource_path=str(Path(self._assets_dir) / Path(data_file)),
             resource_kwargs={},
         )
-        # now discard the start uid, a real one will be added later
-        self._resource_document.pop("run_start")
-        self._asset_docs_cache.append(("resource", self._resource_document))
 
-        data_file = str(
+        self._data_file = str(
             Path(self._resource_document["root"])
             / Path(self._resource_document["resource_path"])
         )
 
+        # now discard the start uid, a real one will be added later
+        self._resource_document.pop("run_start")
+        self._asset_docs_cache.append(("resource", self._resource_document))
+
+        logger.debug(f"{self._data_file = }")
+
+        self._h5file_desc = h5py.File(self._data_file, "w")
```

Comment:
Better to use the exclusive mode (`'x'` if I recall correctly) that will fail if the file you are trying to write to exists.  This is to make sure we do not accidentally over-write older data!